ui/objects/semantic-ui/js/objects/detail/Carousel.jsx

```diff
@@ -0,0 +1,104 @@
+import React, { useState, useEffect } from "react";
+import Slider from "react-slick";
+import {
+  Modal,
+  Image,
+  Button,
+  Grid,
+  Header
+} from "semantic-ui-react";
+
+
+
+export const ImgCarousel = ({ imagesCollection, fileName}) => {
+  const [modalOpen, setModalOpen] = useState(false);
+  const [selectedImageIndex, setSelectedImageIndex] = useState(0);
+  const [slidesToShow, setSlidesToShow] = useState(5);
+  
+
+  const settings = {
```

Comment:
this prop depends on screen width ` useEffect(() => {
    const updateSlidesToShow = () => {
      if (window.innerWidth <= 992 && window.innerWidth >= 530) {
        setSlidesToShow(3);
      } else if (window.innerWidth <= 530) {
        setSlidesToShow(2);
      } else {
        setSlidesToShow(5);
      }
    };

    updateSlidesToShow();
    window.addEventListener("resize", updateSlidesToShow);

    return () => {
      window.removeEventListener("resize", updateSlidesToShow);
    };
  }, []);`

Reply:
Only the `slidesToShow` state depends on screen width, other settings are static. It would be useful, if users of Carousel component could tweak those other settings :)

Reply:
Still an issue

Reply:
built-in settings are `var settings = {
    dots: true,
    infinite: true,
    speed: 500,
    slidesToShow: 1,
    slidesToScroll: 1,
  };` mine : `const settings = {
    infinite: false,
    speed: 100,
    slidesToShow: slidesToShow,
    swipeToSlide: true,
  };`